src/suricata.c

```diff
@@ -1024,6 +1024,10 @@ static TmEcode ParseInterfacesList(const int runmode, char *pcap_dev)
                 SCLogError("No interface found in config for af-packet");
                 SCReturnInt(TM_ECODE_FAILED);
             }
+            int retval = CheckAFPacketIPSDevs();
```

Comment:
can this be checked in the af-packet files instead?

Reply:
Yes. As per your main suggestion since all of the other things done are invalid, it should all probably be there.